src/com/palmergames/bukkit/towny/command/TownCommand.java

```diff
@@ -3601,10 +3617,19 @@ private void townDeposit(Player player, int amount) {
 				TownyMessaging.sendErrorMsg(player, preEvent.getCancelMessage());
 				return;
 			}
-			
-			// Deposit into town.
-			town.depositToBank(resident, amount);
-			
+
+			if(town.getAccount().isBankrupt()) {
+				// Deposit into town.
+				town.depositToBank(resident, amount);
+				//Has bankruptcy been cleared ?
+				if(!town.getAccount().isBankrupt()) {
+					plugin.resetCache(); //Allow perms change to take effect immediately
+					TownyMessaging.sendGlobalMessage(String.format(Translation.of("msg_town_debts_cleared"), town.getFormattedName()));
+				}
+			} else {

```

Comment:
I haven't tested this yet, but I am not certain that this is paying into the debt account.